Game/MyProject/Game/SceneManagement/RoadNode.cpp

```diff
@@ -0,0 +1,77 @@
+
+#include "RoadNode.hpp"
+
+RoadNode::RoadNode(const std::string& name, std::vector<std::pair<IvVector3,IvVector3>> rMarginPoints, std::shared_ptr<SceneNode> graphRoot) :  SceneNode(name),roadPoints(rMarginPoints)
+{
+    root = graphRoot;
```

Comment:
Cyclic dependency. Nodes will never be deleted.
Child nodes should never own root nodes, since root nodes already own child nodes.